slack/src/main/java/org/togglz/slack/SlackWrapperStateRepository.java

```diff
@@ -0,0 +1,16 @@
+package org.togglz.slack;
+
+import org.togglz.core.repository.StateRepository;
+import org.togglz.core.repository.composite.CompositeStateRepository;
+
+/**
+ * @author Tomasz Skowroński
+ * @since 2.4.0
+ */
+public class SlackWrapperStateRepository extends CompositeStateRepository {
```

Comment:
Like this?
```
public class SlackStateRepository implements StateRepository {

    private final StateRepository wrappedRepository;
    private final SlackNotifications slackNotifications;

    public SlackStateRepository(StateRepository wrappedRepository, NotificationConfiguration configuration, UserProvider userProvider) {
        this.wrappedRepository = wrappedRepository;
        this.slackNotifications = new SlackNotifications(configuration, userProvider);
    }

    @Override
    public FeatureState getFeatureState(Feature feature) {
        return wrappedRepository.getFeatureState(feature);
    }

    @Override
    public void setFeatureState(FeatureState featureState) {
        wrappedRepository.setFeatureState(featureState);
        slackNotifications.notify(featureState);
    }
}
```
and package scope SlackNotifications (previously SlackStateRepository)